src/components/account.tsx

```diff
@@ -0,0 +1,196 @@
+    decimals: number;
+}
+interface TokenBalanceAllowance {
+    token: Token;
+    balance: BigNumber;
+    allowance: BigNumber;
+    tradeable: boolean;
+}
+
+interface AccountState {
+    balances: { [address: string]: TokenBalanceAllowance[] };
+    selectedAccount: string;
+}
+
+export class Account extends React.Component<Props, AccountState> {
+    constructor(props: Props) {
+        super(props);
+        this.state = { balances: {}, selectedAccount: '' };
+        void this.fetchAccountDetailsAsync();
+        setInterval(() => {
+            void this.checkAccountChangeAsync();
+        }, 2000);
+    }
+    public fetchAccountDetailsAsync = async () => {
+        const { web3Wrapper, erc20TokenWrapper } = this.props;
+        const { balances } = this.state;
+        const addresses = await web3Wrapper.getAvailableAddressesAsync();
+        const address = addresses[0];
+        if (!address) {
+            return;
+        }
+        const networkId = await web3Wrapper.getNetworkIdAsync();
+        const tokens = tokensByNetwork[networkId];
+        // Fetch all the Balances for all of the tokens
+        const allBalancesAsync = _.map(
+            tokens,
+            async (token: Token): Promise<TokenBalanceAllowance> => {
+                const balance = await erc20TokenWrapper.getBalanceAsync(token.address, address);
+                const allowance = await erc20TokenWrapper.getProxyAllowanceAsync(token.address, address);
+                const numberBalance = new BigNumber(balance);
+                return { token, balance: numberBalance, allowance, tradeable: true };
+            },
+        );
+
+        const results = await Promise.all(allBalancesAsync);
+        balances[address] = results;
+        // Fetch the Balance of Ether
+        const weiBalance = await web3Wrapper.getBalanceInWeiAsync(address);
+        balances[address] = [
+            ...balances[address],
+            {
+                token: ETHER_TOKEN,
+                balance: weiBalance,
+                allowance: new BigNumber(0),
+                tradeable: false,
+            } as TokenBalanceAllowance,
+        ];
+
+        // Update the state in React
+        this.setState(prev => {
+            const prevSelectedAccount = prev.selectedAccount;
+            const selectedAccount = prevSelectedAccount !== address ? address : prevSelectedAccount;
+            return { ...prev, balances, selectedAccount };
+        });
+    }
+    public checkAccountChangeAsync = async () => {
+        const { web3Wrapper } = this.props;
+        const { selectedAccount } = this.state;
+        const addresses = await web3Wrapper.getAvailableAddressesAsync();
+        const address = addresses[0];
+        if (!address) {
+            return;
+        }
+        if (selectedAccount !== address) {
+            const balances = {};
+            this.setState(prev => ({ ...prev, balances, selectedAccount }));
+            void this.fetchAccountDetailsAsync();
```

Comment:
I have a slight preference for adding the TSLint disable rule explicitly, but am open to arguments.